db/migrations/versions/050_create_organisationtype.py

```diff
@@ -0,0 +1,57 @@
+"""create_OrganisationType
+
+Revision ID: 050_create_organisationtype
+Revises: 049_add_external_reference_code
+Create Date: 2024-11-07 10:09:03.116129
+
+"""
+
+import uuid
+
+import sqlalchemy as sa
+from alembic import op
+
+import data_store
+
+# revision identifiers, used by Alembic.
+revision = "050_create_organisationtype"
+down_revision = "049_add_external_reference_code"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "organisation_type_dim",
+        sa.Column("type_name", sa.String(), nullable=False),
+        sa.Column("id", data_store.db.types.GUID(), nullable=False),
+        sa.PrimaryKeyConstraint("id", name=op.f("pk_organisation_type_dim")),
+        sa.UniqueConstraint("type_name", name=op.f("uq_organisation_type_dim_type_name")),
+    )
+    with op.batch_alter_table("organisation_dim", schema=None) as batch_op:
+        batch_op.add_column(sa.Column("organisation_type_id", data_store.db.types.GUID(), nullable=True))
+        batch_op.create_foreign_key(
+            batch_op.f("fk_organisation_dim_organisation_type_id_organisation_type_dim"),
+            "organisation_type_dim",
+            ["organisation_type_id"],
+            ["id"],
+            ondelete="SET NULL",
+        )
+
+    # Insert "Local Authority" into organisation_type_dim
+    new_id = str(uuid.uuid4())
+    op.execute(f"INSERT INTO organisation_type_dim (id, type_name) VALUES ('{new_id}', 'Local Authority')")
```

Comment:
There has been some discussion before around doing data migrations as part of alembic migrations. Alembic is for structural changes and not really designed for inserting arbitrary "reference" data (which is what we use it for), which is why we have things like the db seeding scripts/functions.

This should probably live there.

That said I agree with Marc that a whole separate table for this feels like overkill compared to an enum...